src/persist/src/indexed/mod.rs

```diff
@@ -282,6 +344,10 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
         let mut updates_by_id: HashMap<Id, Vec<(SeqNo, (Vec<u8>, Vec<u8>), u64, isize)>> =
             HashMap::new();
         let desc = self.buf.snapshot(|seqno, buf| {
+            // Skip updates not in advance of the last sequence number we have observed.
```

Comment:
elaborate on why, please. they previously made it into the future, but the buffer truncation failed, i think?